src/librustc_data_structures/bit_set.rs

```diff
@@ -341,21 +342,26 @@ impl<T: Idx> SparseBitSet<T> {
     }
 
     fn insert(&mut self, elem: T) -> bool {
-        // Ensure there are no duplicates.
-        if self.0.contains(&elem) {
-            false
+        assert!(self.len() < SPARSE_MAX);
+        if let Some(i) = self.0.iter().position(|&e| e >= elem) {
```

Comment:
I guess it doesn't seem worth using [`binary_search`](https://doc.rust-lang.org/std/primitive.slice.html#method.binary_search) for this?

Reply:
Nah, only `8` elements...

Reply:
Yeah, I haven't compared linear vs. binary, but I haven't seen evidence that this code is hot. I bet the difference is minimal, and if anything favours linear with that few elements.